apps/server/src/services/delayUtils.ts

```diff
@@ -0,0 +1,38 @@
+import { OntimeEvent, OntimeRundown, SupportedEvent } from 'ontime-types';
+
+import { deleteAtIndex } from '../utils/arrayUtils.js';
+
+export function _applyDelay(eventId: string, rundown: OntimeRundown): OntimeRundown {
+  const delayIndex = rundown.findIndex((event) => event.id === eventId);
+  const delayEvent = rundown.at(delayIndex);
+
+  if (delayEvent.type !== SupportedEvent.Delay) {
+    throw new Error('Given event ID is not a delay');
+  }
+
+  const updatedRundown = [...rundown];
+  const delayValue = delayEvent.duration;
+
+  if (delayValue === 0 || delayIndex === rundown.length - 1) {
+    // nothing to apply
+    return updatedRundown;
+  }
+
+  for (let i = delayIndex + 1; i < rundown.length; i++) {
+    const currentEvent = updatedRundown[i];
+
+    if (currentEvent.type === SupportedEvent.Block) {
+      break;
+    } else if (currentEvent.type === SupportedEvent.Event) {
+      const event = currentEvent as OntimeEvent;
```

Comment:
What's the point of this reassignment? If you're trying to create a copy, then you should spread `{ ...oldObj }`

Reply:
thats correct. The reason for reassigning was to make typescript happy since it is handy having good typing in the following lines. 
It was a hack that stayed behind, I guess the only way is to have a function that asserts a type with the `is` keyword?

Reply:
I'm having déjà vu right now — I believe I commented something similar somewhere else. It doesn't work without `as`? I believe the condition should be intelligent enough to narrow down the type.